port/unix/omrsysinfo.c

```diff
@@ -3722,24 +3744,44 @@ isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
 		}
 
 		while (0 == feof(cgroupFile)) {
+			char buffer[PATH_MAX + 1024];
 			char cgroup[PATH_MAX];
 			char subsystems[1024];
 			int32_t hierId = -1;
 
-			rc = fscanf(cgroupFile, PROC_PID_CGROUP_ENTRY_FORMAT, &hierId, subsystems, cgroup);
+			fgets(buffer, PATH_MAX + 1024, cgroupFile);
+			if(0 != ferror(cgroupFile)){
+				int32_t osErrCode = errno;
+				Trc_PRT_isRunningInContainer_fgets_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "fgets failed to read %s file stream with errno=%d", OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				goto _end;
+			} else if (NULL == buffer) {
+				break;
+			}
+			rc = sscanf(buffer, PROC_PID_CGROUP_ENTRY_FORMAT, &hierId, subsystems, cgroup);
 			/* Ensure we didn't overflow */
 			Assert_PRT_true(strlen(subsystems) < 1024);
-			Assert_PRT_true(strlen(cgroup) < PATH_MAX);
+			Assert_PRT_true(strlen(cgroup) < PATH_MAX);		
 
 			if (EOF == rc) {
 				break;
+			} else if (1 == rc) {
+				rc = sscanf(buffer, PROC_PID_CGROUP_SYSTEMD_ENTRY_FORMAT, &hierId, cgroup);
+				Assert_PRT_true(strlen(cgroup) < PATH_MAX); 
+
+				if (2 != rc) {
+					Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
+					rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
+					goto _end;
+				}
+				subsystems[0] = '\0';
```

Comment:
why is subsystems being modified?

Reply:
If it gets to this point, it means subsystems is empty - this is to overwrite the subsystem from the previous iteration. Subsystems is not being used here, but I left it in since I don't know how this function may be changed in the future. Should I remove this for the sake of efficiency?

Reply:
If it is not required it would be great if it could be removed.